p2p/src/peer.rs

```diff
@@ -173,9 +169,12 @@ impl Peer {
 		false
 	}
 
-	/// Whether this peer is still connected.
+	/// Whether this peer is currently connected.
 	pub fn is_connected(&self) -> bool {
-		self.check_connection()
+		if self.connection.is_none() {
```

Comment:
do we set `connection` to None somewhere?

Reply:
I believe there is a small period of time during the initialization of a new peer connection where we instantiate the peer instance but we do not yet have a live connection.
I don't think we ever set a connection back to None once a peer is up and running.

Reply:
Related - #2801 (get rid of the option wrapping the peer connection).
The cleanup in #2801 is probably worth doing regardless of what we decide here.